sqlnexus/PerfStatsAnalysis.sql

```diff
@@ -3478,6 +3485,46 @@ END
 GO
 
 
+
+/**************************************************************************************************
+owner:  JAMGRIF
+date: 8/23/22
+description: Adds check and warning if CAL license is in use and schedulers are less than CPUs 
+
+***************************************************************************************************/
+
+create procedure usp_CalvsCore
+AS
+IF ((OBJECT_ID ('tbl_ServerProperties') IS NOT NULL) and (OBJECT_ID ('tbl_AnalysisSummary') IS NOT NULL))
+BEGIN
+	DECLARE @schedCount int = 0
+	DECLARE @cpuCount int = 0
+	--make sure there is only 1 distinct row (this process might get more than 1 server...probably FCI situtation when it does)
+	DECLARE @cpuRows int = 0;
+	--unless there is more than 1 row
+	SELECT DISTINCT @cpuRows = COUNT(PropertyValue)
+	FROM [tbl_ServerProperties] WHERE PropertyName in ('cpu_count')
+	IF (@cpuRows = 1) --then we are OK to use the number
+	BEGIN
+		SELECT DISTINCT @cpuCount = PropertyValue
+		FROM [tbl_ServerProperties] WHERE PropertyName in ('cpu_count')
+		SELECT DISTINCT @schedCount = PropertyValue
+		FROM [tbl_ServerProperties] WHERE PropertyName in ('number of visible schedulers')
+	END
+
+	IF @cpuCount <> 0 OR @cpuCount is not null
+	BEGIN
+		IF EXISTS (SELECT 1 FROM [dbo].[tbl_ServerProperties] where PropertyName = 'Edition' AND PropertyValue NOT LIKE '%Core-based%') AND @cpuCount > @schedCount 
+			BEGIN
+				UPDATE [dbo].[tbl_AnalysisSummary]
+				SET [Status] = 1,
+				[Description] = 'Customer is using CAL license and CPUs (' + CONVERT(VARCHAR, @cpuCount) + ') are greater than schedulers (' + CONVERT(VARCHAR, @schedCount) + ') online, check the errorlog to confirm. Customer could benefit by upgrading to CORE license.'
```

Comment:
@jagriffin5 , keep in mind that this is a public product and audience is everyone. So changing the verbiage a bit. If you like the proposed verbiage, you can just a accept/commit the suggestion.

```suggestion
				[Description] = 'Using CAL license and CPUs (' + CONVERT(VARCHAR, @cpuCount) + ') are greater than schedulers (' + CONVERT(VARCHAR, @schedCount) + ') online, check the errorlog to confirm. The user is not taking full advantage of available hardware and can benefit from upgrading to CORE license.'
```